packages/core/test/financial-templates/common/FundingRateApplier.js

```diff
@@ -196,11 +196,32 @@ contract("FundingRateApplier", function(accounts) {
     assert.equal(test4[1].rawValue, toWei("0"));
   });
 
-  it("Initial 0 funding rate", async () => {
+  it("Initial funding rate struct is correct", async () => {
+    const fundingRate = await fundingRateApplier.fundingRate();
+    assert.equal(fundingRate.rate.toString(), "0");
+    assert.equal(hexToUtf8(fundingRate.identifier), hexToUtf8(identifier));
+    assert.equal(fundingRate.cumulativeMultiplier.toString(), toWei("1"));
+    assert.equal(fundingRate.updateTime.toString(), startTime.toString());
+    assert.equal(fundingRate.applicationTime.toString(), startTime.toString());
+    assert.equal(fundingRate.proposalTime.toString(), "0");
+  });
+  it("Calling applyFundingRate without a pending proposal does not change multiplier", async () => {
     await fundingRateApplier.setCurrentTime(startTime + 1000);
-    await fundingRateApplier.applyFundingRate();
+    const receipt = await fundingRateApplier.applyFundingRate();
     assert.equal((await fundingRateApplier.fundingRate()).cumulativeMultiplier.rawValue.toString(), toWei("1"));
     assert.equal((await fundingRateApplier.fundingRate()).rate.rawValue.toString(), "0");
+
+    // A NewFundingRateMultiplier is emitted.
+    truffleAssert.eventEmitted(receipt, "NewFundingRateMultiplier", ev => {
```

Comment:
This test is important because it makes it obvious that `NewFundingRateMultiplier` is emitted even if the funding rate is unchanged. Open question: Feature or bug?

Reply:
IMO, this event is really just saying "FundingRateApplied", so it's kinda a weird name (even though it is technically changing the funding rate multiplier).

It gets emitted effectively every time the contract is touched. Does that mean it's not useful and a bad use of users' gas? I could see that argument.

Another question is whether we should have an event for when the funding rate is updated after the getPrice method returns. Something like "FundingRateUpdated" that just gives the new funding rate?

IMO, we should weigh these concerns against the cost of making changes this late.

Thoughts?

Reply:
I don't think its a bug as much as an artifact of bad naming. I do think we should deprecate `NewFundingRateMultiplier` in favor of a `FundingRateUpdated` event and will address in a follow up PR: #2282